src/store/modules/collection.ts

```diff
@@ -19,20 +19,22 @@ class Collection extends VuexModule {
   public page = 0;
 
   get getPosts() {
-    if (this.items) {
-      const postTypeItems = this.items.filter($ => $.type in PostTypes);
-
-      const activityItems = this.items
-        .filter($ => !($.type in PostTypes))
-        .filter(
-          ($: Activity) =>
-            !!$.object &&
-            ActivityObjectHelper.hasProperty($.object, "type") &&
-            ($.object as ActivityObject).type in PostTypes
-        )
-        .map(ActivityObjectHelper.extractObjectFromActivity);
-      return postTypeItems.concat(activityItems);
+    if (!this.items) {
+      return;
     }
+
+    const postTypeItems = this.items.filter($ => $.type in PostTypes);
+
+    const activityItems = this.items
+      .filter($ => !($.type in PostTypes))
+      .filter(
+        ($: ActivityImplementation) =>
+          !!$.object &&
+          ActivityObjectHelper.hasProperty($.object, "type") &&
+          ($.object as ActivityObject).type in PostTypes
+      )
```

Comment:
still don't get what this filter is for...

Reply:
I check from the collection if it's an activity like Create, Delete, Update and Undo

`!!$.object &&`
Here I look if the activity is set to the property object

`ActivityObjectHelper.hasProperty($.object, "type") &&`
This line tests whether the object has the Type property set so that it is not a link

`($.object as ActivityObject).type in PostTypes`
Here you can see if the type is a post type like an article or a note